src/server/helpers/middleware.js

```diff
@@ -211,3 +211,34 @@ export function makeCollectionLoader() {
 		}
 	};
 }
+
+export async function validateCollectionParams(req, res, next) {
+	const {collaborators = [], name, entityType} = req.body;
+	const {orm} = req.app.locals;
+	const {Editor} = orm;
+
+	if (!_.trim(name).length) {
+		return next(new error.BadRequestError('Invalid collection name: Empty string not allowed', req));
+	}
+	const entityTypes = _.keys(utils.getEntityModels(orm));
+	if (!entityTypes.includes(entityType)) {
+		return next(new error.BadRequestError(`Invalid entity type: ${entityType} does not exist`, req));
+	}
+	const collaboratorIds = collaborators.map(collaborator => collaborator.id);
+	for (let i = 0; i < collaboratorIds.length; i++) {
+		const collaboratorId = collaboratorIds[i];
+		if (!Number.isInteger(collaboratorId) || collaboratorId <= 0) {
```

Comment:
I'm worried that this might misfire for strings, considering `Number.isInteger("5") => false`.
Number manipulation is always a PITA in javascript…

For this specific case where we want to ascertain it's a positive integer, the easiest way I know out of this connundrum is a simple regexp: `/^\d+$/.test("1234")` which will work for strings and numbers alike and allow nothing other than positive integers, with nothing before or after.